### PR TITLE
BaSP-MR-132: Admin profile

### PR DESCRIPTION
**Pull request by victoriassol**
Created an admin profile where you're able to edit their information and delete the profile.
The profile grabs the first admin on the list and showcases that data.
Implemented react hook forms library and absolute imports.

### CODE REVIEW

File: src/Components/Admins/Form/index.jsx [hunk 1]
```diff
@@ -1,31 +1,45 @@
...
+import Button from 'Components/Shared/Button';
+import ConfirmModal from 'Components/Shared/ConfirmModal';
+import ResponseModal from 'Components/Shared/ResponseModal';
+import { Input } from 'Components/Shared/Inputs';
+import { addAdmin, getAdminsById, editAdmin } from 'Redux/Admins/thunks';
 import { useDispatch, useSelector } from 'react-redux';
 import { handleDisplayToast } from '../../../Redux/Shared/ResponseToast/actions';
 import { resetState } from '../../../Redux/Admins/actions';
+import { useForm } from 'react-hook-form';
+import adminSchema from '../../../Validations/admin';
```
Comment (josedaniele) [inline src/Components/Admins/Form/index.jsx]: please apply absolute paths

File: src/Components/Admins/Profile/index.jsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+import { Input } from 'Components/Shared/Inputs';
+import ConfirmModal from 'Components/Shared/ConfirmModal';
+import ResponseModal from 'Components/Shared/ResponseModal';
+import { getAdmins, editAdmin, deleteAdmin } from 'Redux/Admins/thunks';
+import { handleDisplayToast } from 'Redux/Shared/ResponseToast/actions';
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import adminSchema from '../../../Validations/admin';
```
Comment (josedaniele) [inline src/Components/Admins/Profile/index.jsx]: same here

**Review by josedaniele — CHANGES_REQUESTED** [review]
note that the spinner of the loader is off center, fix that.
Everything else works perfectly, good job.

**Review by josedaniele — APPROVED** [review]

File: src/Components/Admins/Profile/index.jsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    setValue('phone', defaultAdmin.phone);
+    setValue('email', defaultAdmin.email);
+    setValue('city', defaultAdmin.city);
+    setValue('password', defaultAdmin.password);
+    clearErrors();
+  };
+
+  return (
+    <div className={styles.form}>
+      {defaultAdmin == {} && <p>There are no admins</p>}
```
Comment (WalterArielPavicich) [inline src/Components/Admins/Profile/index.jsx]: I think that the empty object couldn't be evaluated this way, I found this way that works!

```suggestion
      {Object.keys(defaultAdmin).length === 0 && <p>There are no admins</p>}
```

File: src/Components/Admins/Profile/profile.module.css [hunk 4]
```diff
@@ -0,0 +1,81 @@
+.content {
+  display: flex;
+  flex-direction: column;
+  background-color: #f4f4f4;
+  margin: 10% auto;
+  margin-top: 30px;
+  height: 100%;
+  width: 70%;
+  border-radius: 7px;
+  box-shadow: 0 5px 8px 0 rgba(0, 0, 0, 0.2), 0 7px 20px 0 rgba(0, 0, 0, 0.17);
```
Comment (WalterArielPavicich) [inline src/Components/Admins/Profile/profile.module.css]: could change rgba to hex

File: src/Redux/Admins/actions.js [hunk 5]
```diff
@@ -17,6 +17,13 @@ import {
   RESET_STATE
 } from './constants';
 
+export const setDefaultAdmin = (admin) => {
+  return {
+    type: 'SET_DEFAULT_ADMIN',
```
Comment (WalterArielPavicich) [inline src/Redux/Admins/actions.js]: could be added to constants.js file and imported from there
```suggestion
    type: SET_DEFAULT_ADMIN,
```

File: src/Redux/Admins/reducers.js [hunk 6]
```diff
@@ -19,13 +19,20 @@ import {
...
   data: [],
+  defaultAdmin: {},
   isPending: false,
   error: null,
   success: false
 };
 
 const reducers = (state = INITIAL_STATE, action) => {
   switch (action.type) {
+    case 'SET_DEFAULT_ADMIN': {
```
Comment (WalterArielPavicich) [inline src/Redux/Admins/reducers.js]: same here

```suggestion
    case SET_DEFAULT_ADMIN: {
```

File: src/Components/Admins/Profile/index.jsx (hunk 3 above)
Comment (victoriassol) [reply src/Components/Admins/Profile/index.jsx]: awesome, thank you!

**Review by WalterArielPavicich — APPROVED** [review]
Everything works very well! Great job!

**Comment by gabiholl** [comment]
Its fixed! works fine!

**Pull request merged**